recipes/newrelic/infrastructure/cloud/aws-linux.yml

```diff
@@ -39,7 +132,112 @@ install:
     setup:
       cmds:
         - |
-          exit 131
+          # Constants
+          roleName="NewRelicInfrastructure-Integrations"
+
+          policy=$(cat <<EOT
+          {
+              "Version": "2012-10-17",
+              "Statement": [
+                  {
+                      "Sid": "",
+                      "Effect": "Allow",
+                      "Principal": {
+                          "AWS": "arn:aws:iam::754728514883:root"
+                      },
+                      "Action": "sts:AssumeRole",
+                      "Condition": {
+                          "StringEquals": {
+                              "sts:ExternalId": "{{.NEW_RELIC_ACCOUNT_ID}}"
+                          }
+                      }
+                  }
+              ]
+          }
+          EOT
+          )
+
+          createRole=$(aws iam create-role --role-name $roleName --assume-role-policy-document file://<(echo $policy) 2>&1)
+
+          getRole=$(aws iam get-role --role-name $roleName 2>&1)
+
+          if [ "$(echo "$getRole" | grep "^An error occurred (NoSuchEntity)*" | wc -l)" -eq 1 ] ; then
+              echo "$createRole" >&2
```

Comment:
@noahmmcgivern it's a good idea to assert the role exists. However we'll need more user guidance here if for some reason the role can't be created. Maybe refer the user to the doc for creating the role themselves (or have someone do it for the user).